Description: Disable SONAMEs for all libraries
 Since all the libraries are private, they should not have SONAMEs
Author: James Cowgill <james410@cowgill.org.uk>
Last-Update: 2014-06-06
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -499,5 +499,21 @@ include(plugin)
 ## Scan for user plugins
 CL_SCAN_FOR_PLUGINS()
 
+# Disable SONAMES on all libraries (since none are public)
+set_target_properties(libcodelite plugin wxshapeframework databaselayersqlite
+    abbreviation CallGraph ContinuousBuild DebuggerGDB Wizards Outline QMakePlugin
+    UnitTestsPP cppchecker cscope wxFormBuilder Tweaks CodeFormatter Copyright
+    DatabaseExplorer ExternalTools SnipWiz Subversion ZoomNavigator git
+    CMakePlugin CodeLiteDiff MemCheck
+    PROPERTIES NO_SONAME TRUE)
+
+if (WITH_SFTP)
+    set_target_properties(SFTP PROPERTIES NO_SONAME TRUE)
+endif(WITH_SFTP)
+
+if (WITH_LLDB)
+    set_target_properties(LLDBDebugger PROPERTIES NO_SONAME TRUE)
+endif()
+
 # Don't unset this earlier: it would affect the plugins' CMakeLists.txts
 unset(PREFIX CACHE)
